contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1562,15 +1562,32 @@ is_alter_server_stmt(GrantRoleStmt *stmt)
 	{
 		RoleSpec   *spec = (RoleSpec *) linitial(stmt->granted_roles);
 
-		if (strcmp(spec->rolename, "sysadmin") != 0)	/* only supported server
+		if (strcmp(spec->rolename, "sysadmin") == 0)	/* only supported server
 														 * role */
-			return false;
+			return true;
 	}
-	/* has one and only one grantee  */
-	if (list_length(stmt->grantee_roles) != 1)
-		return false;
 
-	return true;
+	return false;
+}
+
+bool
+is_grant_sysadmin_stmt(GrantRoleStmt *stmt)
```

Comment:
this looks similar with is_alter_server_stmt(). Let's modify the existing is_alter_server_stmt() function only such that it can used for both dialects.